base/docs/Docs.jl

```diff
@@ -509,6 +509,8 @@ function docm(source::LineNumberNode, mod::Module, ex)
         return REPL.lookup_doc(ex)
     end
 end
+# Drop incorrect line numbers produced by nested macro calls.
+docm(source::LineNumberNode, mod::Module, _, _, x...) = docm(source, mod, x...)
```

Comment:
Just out of curiousity: when and where does this signature actually get called?

Reply:
It's called indirectly by the `__doc__!` method above since it generates macro calls to `@doc` with two sets of source/module info, this strips out the incorrect ones and passes back to `docm`. Took me quite a while to debug my way through it initially.